lib/server.go

```diff
@@ -1400,6 +1415,27 @@ func (srv *Server) _handleSnapshot(pp *Peer, msg *MsgDeSoSnapshotData) {
 	srv.GetBlocks(pp, int(headerTip.Height))
 }
 
+// updateDbOpts closes the current badger DB instance and re-opens it with the provided new options.
+func (srv *Server) updateDbOpts(opts badger.Options) {
+	// Make sure that a mempool process doesn't try to access the DB while we're closing and re-opening it.
+	srv.mempool.mtx.RLock()
+	defer srv.mempool.mtx.RUnlock()
+	// Make sure that a server process doesn't try to access the DB while we're closing and re-opening it.
+	srv.DbMutex.Lock()
+	defer srv.DbMutex.Unlock()
```

Comment:
Hmm.. I'm a bit confused. I think DbMutex is only used here, which means that if anything else has a handle on the db then it won't exclude it from reading from that handle. So does this really do anything? It seems like the only thing it prevents is calling updateDbOpts in multiple threads, but nothing else.

Reply:
This is utilized in backend in a few recurring jobs, such as the hot feed routine.

Reply:
Hmm I see it's used in backend.